programs/odbc-bridge/ODBCConnectionFactory.h

```diff
@@ -81,8 +81,12 @@ T execute(nanodbc::ConnectionHolderPtr connection_holder, std::function<T(nanodb
     }
     catch (const nanodbc::database_error & e)
     {
-        /// SQLState, connection related errors start with 08S0.
-        if (e.state().starts_with("08S0"))
+        tryLogCurrentException(__PRETTY_FUNCTION__);
+        /// SQLState, connection related errors start with 08 (main: 08S01), cursor invalid state is 24000.
+        /// Invalid cursor state is a retriable error.
+        /// Invalid transaction state 25000. Truncate to 4 letters, because nanodbc does it.
+        /// https://docs.microsoft.com/ru-ru/sql/odbc/reference/appendixes/appendix-a-odbc-error-codes?view=sql-server-ver15
+        if (e.state().starts_with("08") || e.state().starts_with("24") || e.state().starts_with("2500"))
```

Comment:
```suggestion
        if (e.state().starts_with("08") || e.state().starts_with("24") || e.state().starts_with("25"))
```
See https://www.postgresql.org/docs/10/errcodes-appendix.html